src/uct/ib/ud/accel/ud_mlx5.c

```diff
@@ -744,7 +744,9 @@ static ucs_status_t uct_ud_mlx5_iface_create_qp(uct_ib_iface_t *ib_iface,
     return status;
 
 err_destroy_qp:
-    uct_ib_mlx5_destroy_qp(ib_md, qp);
+    if (qp->verbs.qp) {
```

Comment:
do not go to `err_destroy_qp` in case of `uct_ib_mlx5_iface_create_qp` failed?
and we should go here if `uct_ib_mlx5_txwq_init` failed.

Reply:
that's right, seems wrong goto

Reply:
@dmitrygx 
Besides correcting the ```wrong goto```,
should it release ```resource domain``` when failing at [uct_ib_iface_create_qp](https://github.com/openucx/ucx/blob/v1.12.1/src/uct/ib/mlx5/ib_mlx5.c#L304)? the resource domain is got at calling [uct_ib_mlx5_iface_fill_attr](https://github.com/openucx/ucx/blob/v1.12.1/src/uct/ib/mlx5/ib_mlx5.c#L296)

Reply:
if it's created inside uct_ib_mlx5_iface_create_qp, it should already be released inside uct_ib_mlx5_iface_create_qp

Reply:
@yosefe 
The current code doesn't release it https://github.com/openucx/ucx/blob/v1.12.1/src/uct/ib/mlx5/ib_mlx5.c#L290-L309
I also checked other places which use ```uct_ib_mlx5_iface_fill_attr``` to get ```resource domain```, they also doesn't release the ```resource domain``` at failure case.
1. https://github.com/openucx/ucx/blob/v1.12.1/src/uct/ib/dc/dc_mlx5.c#L330
2. https://github.com/openucx/ucx/blob/v1.12.1/src/uct/ib/rc/accel/rc_mlx5_iface.c#L313

Reply:
seems like a bug, @Artemy-Mellanox do you agree?

Reply:
Absolutely. Missing uct_ib_mlx5_iface_put_res_domain() call in failure flow in all three cases.

Reply:
I'll add them in this PR.